impeller/entity/contents/filters/morphology_filter_contents.cc

```diff
@@ -93,7 +93,7 @@ std::optional<Entity> DirectionalMorphologyFilterContents::RenderFilter(
     frame_info.texture_sampler_y_coord_scale =
         input_snapshot->texture->GetYCoordScale();
 
-    auto transform = entity.GetTransformation() * effect_transform;
+    auto transform = entity.GetTransformation() * effect_transform.Basis();
```

Comment:
> The given effect transform should already have its translation/projective coords stripped. Can we remove the `Basis()` calls here and in the gaussian blur?

I modified the implementation of `FilterContents::SetEffectTransform`, because MatrixImageFilter needs an original matrix instead of a stripped one, otherwise the result will be incorrect (such as when nesting `Transformed` widget). So I added `Basis()` to blur and morphology.

``` diff
void FilterContents::SetEffectTransform(Matrix effect_transform) {
-  effect_transform_ = effect_transform.Basis();
+  effect_transform_ = effect_transform;
}
```